src/main/java/org/candlepin/model/Consumer.java

```diff
@@ -1010,26 +1010,21 @@ public Consumer setEnvironmentIds(List<String> environmentIds) {
             this.environmentIds.clear();
         }
 
-        // Creating a new map instance and assigning it to the environmentIds.
-        // This ensures that, we are deleting the existing entries and inserting the new entries.
-        // This is done to avoid constraint (cp_consumer_environments_pkey) violation exception.
-        // Whenever we try to update the existing data (changing priority), it actually updates
-        // the environment Id and not the priority.
-        LinkedHashMap<String, String> envMap = new LinkedHashMap<>();
         if (environmentIds != null) {
-
-            for (String environmentId : environmentIds) {
-                if (envMap.containsValue(environmentId)) {
-                    throw new DuplicateEntryException("Environment " + environmentId +
-                        " specified more than once.");
-                }
-            }
```

Comment:
We need to iterate twice here, so I don't think we should be removing this. See here for explanation: https://github.com/candlepin/candlepin/pull/3129#discussion_r722542881 If anything, we might need to add an extra comment here to explain why 2 loops are needed

Reply:
There was also a related discussion about adding a composite unique key on the consumerId+environmentId columns, so that we don't have to resort to manually checking for duplicates, but that didn't work out because hibernate likes to add new environmentIds as duplicates before removing the old ones during the re-prioritization use case.